apple/internal/partials/swift_static_framework.bzl

```diff
@@ -0,0 +1,114 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Partial implementation for Swift static frameworks."""
+
+load(
+    "@build_bazel_rules_apple//apple/internal:bundling_support.bzl",
+    "bundling_support",
+)
+load(
+    "@build_bazel_rules_apple//apple/internal:file_support.bzl",
+    "file_support",
+)
+load(
+    "@build_bazel_rules_apple//apple/internal:intermediates.bzl",
+    "intermediates",
+)
+load(
+    "@build_bazel_rules_apple//apple/internal:processor.bzl",
+    "processor",
+)
+load(
+    "@bazel_skylib//lib:partial.bzl",
+    "partial",
+)
+load(
+    "@bazel_skylib//lib:paths.bzl",
+    "paths",
+)
+
+def _modulemap_contents(module_name):
+    """Returns the contents for the modulemap file for the framework."""
+    return """\
+framework module {module_name} {{
+  header "{module_name}.h"
+  requires objc
+}}
+""".format(module_name = module_name)
+
+def _swift_static_framework_partial_impl(ctx, swift_static_framework_info):
+    """Implementation for the Swift static framework processing partial."""
+
+    expected_module_name = bundling_support.bundle_name(ctx)
+    if expected_module_name != swift_static_framework_info.module_name:
+        fail("""
+error: Found swift_library with module name {actual} but expected {expected}. Swift static \
+frameworks expect a single swift_library dependency with `module_name` set to the same \
+`bundle_name` as the static framework target.\
+""")
+
+    generated_header = swift_static_framework_info.generated_header
+    swiftdocs = swift_static_framework_info.swiftdocs
+    swiftinterfaces = swift_static_framework_info.swiftinterfaces
+
+    bundle_files = []
+    modules_parent = paths.join("Modules", "{}.swiftmodule".format(expected_module_name))
+
+    for arch, swiftinterface in swiftinterfaces.items():
+        bundle_interface = intermediates.file(
+            ctx.actions,
+            ctx.label.name,
+            "{}.swiftinterface".format(arch),
+        )
+        file_support.symlink(ctx, swiftinterface, bundle_interface)
+        bundle_files.append((processor.location.bundle, modules_parent, depset([bundle_interface])))
+
+    for arch, swiftdoc in swiftdocs.items():
+        bundle_doc = intermediates.file(ctx.actions, ctx.label.name, "{}.swiftdoc".format(arch))
+        file_support.symlink(ctx, swiftdoc, bundle_doc)
+        bundle_files.append((processor.location.bundle, modules_parent, depset([bundle_doc])))
+
+    bundle_header = intermediates.file(
+        ctx.actions,
+        ctx.label.name,
+        "{}.h".format(expected_module_name),
+    )
+    file_support.symlink(ctx, generated_header, bundle_header)
+    bundle_files.append((processor.location.bundle, "Headers", depset([bundle_header])))
+
+    modulemap = intermediates.file(ctx.actions, ctx.label.name, "module.modulemap")
+    ctx.actions.write(modulemap, _modulemap_contents(expected_module_name))
```

Comment:
It would be nice if this wasn't required since it's not needed if you don't depend on this from objc

Reply:
removed modulemaps being required, they will be generated only if the header exists